crates/iceberg/src/catalog/metadata_location.rs

```diff
@@ -15,38 +15,91 @@
 // specific language governing permissions and limitations
 // under the License.
 
+use std::collections::HashMap;
 use std::fmt::Display;
 use std::str::FromStr;
 
 use uuid::Uuid;
 
+use crate::spec::TableProperties;
 use crate::{Error, ErrorKind, Result};
 
 /// Helper for parsing a location of the format: `<location>/metadata/<version>-<uuid>.metadata.json`
+/// or with compression: `<location>/metadata/<version>-<uuid>.gz.metadata.json`
 #[derive(Clone, Debug, PartialEq)]
 pub struct MetadataLocation {
     table_location: String,
     version: i32,
     id: Uuid,
+    compression_suffix: Option<String>,
 }
 
 impl MetadataLocation {
+    /// Determines the compression suffix from table properties.
+    fn compression_suffix_from_properties(properties: &HashMap<String, String>) -> Option<String> {
+        properties
+            .get(TableProperties::PROPERTY_METADATA_COMPRESSION_CODEC)
+            .and_then(|codec| match codec.to_lowercase().as_str() {
+                "gzip" => Some(".gz".to_string()),
+                "none" | "" => None,
+                _ => None,
+            })
+    }
+
     /// Creates a completely new metadata location starting at version 0.
     /// Only used for creating a new table. For updates, see `with_next_version`.
+    #[deprecated(
+        since = "0.8.0",
+        note = "Use new_with_properties instead to properly handle compression settings"
+    )]
     pub fn new_with_table_location(table_location: impl ToString) -> Self {
         Self {
             table_location: table_location.to_string(),
             version: 0,
             id: Uuid::new_v4(),
+            compression_suffix: None,
+        }
+    }
+
+    /// Creates a completely new metadata location starting at version 0,
+    /// with compression settings from the table's properties.
+    /// Only used for creating a new table. For updates, see `with_next_version`.
+    pub fn new_with_properties(
```

Comment:
```suggestion
    pub fn new_table(table: &Table)
```

We may do refactoring to table properties, and we may need other

Reply:
Hi, @emkornfield changing the input param to take a `&TableMetadata` sounds good to me.

Reply:
done.